ppa/archive/admin.py

```diff
@@ -114,6 +116,32 @@ def bulk_add_collection(self, request, queryset):
                                              'to collections')
     bulk_add_collection.allowed_permissions = ('change',)
 
+    def suppress_works(self, request, queryset):
+        # set status to suppressed for every item in the queryset
+        # that is not already suppressed
+        non_suppressed = queryset.exclude(status=DigitizedWork.SUPPRESSED)
+        # save the list of ids being suppressed to update the index after
+        ids_to_suppress = list(non_suppressed.values_list('source_id', flat=True))
+        # change statu in the database
+        updated = non_suppressed.update(status=DigitizedWork.SUPPRESSED)
+        # queryset.update does not trigger save signals;
+        # clear suppressed page + work content from the index
+        solr = SolrClient()
```

Comment:
idk what goes on behind the scenes here, and it might not be expensive at all, but it seems like you could choose to only initialize the client if there are any ids to suppress (below line 131)

Reply:
good point! probably not very expensive, but does require looking up solr connection parameters in django settings, so nice to skip it if we don't need it